tensorboard/webapp/widgets/line_chart_v2/lib/worker/worker_chart.ts

```diff
@@ -0,0 +1,138 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+import {Chart, ChartCallbacks, ChartOption} from '../chart_types';
+import {
+  DataSeries,
+  DataSeriesMetadataMap,
+  Dimension,
+  Extent,
+  ScaleType,
+} from '../internal_types';
+import {
+  GuestToMainMessage,
+  GuestToMainType,
+  InitMessage,
+  MainToGuestEvent,
+  MainToGuestMessage,
+  RendererType,
+} from './worker_chart_types';
+import {compactDataSeries} from './worker_chart_utils';
+import {WorkerPool, WorkerProxy} from './worker_pool';
+
+export class WorkerChart implements Chart {
+  private readonly txMessagePort: MessagePort;
+  private readonly callbacks: ChartCallbacks;
+  private readonly workerInstance: WorkerProxy;
+
+  static readonly workerPool = new WorkerPool('chart_worker.js');
+
+  constructor(option: ChartOption) {
+    this.callbacks = option.callbacks;
+
+    if (option.type === RendererType.SVG) {
+      throw new RangeError('Cannot use SVG for the offscreen line chart.');
```

Comment:
nit: Maybe `!== RendererType.WEBGL` is more appropriate, in case more
renderer types are added?